WordPress/block-development-examples
# Live demo in Basic Block tutorial – redirect to a post with a live block example

The current Live Demo link redirects to the plugins page, which isn't very helpful to a new person trying to interact with the block. This PR updates the link so that it redirects the user to an actual live example.

The Blueprint used in this PR is:

```json
{
  "landingPage": "/wp-admin/post.php?post=4&action=edit",
  "steps": [
    {
      "step": "login",
      "username": "admin",
      "password": "password"
    },
    {
      "step": "mkdir",
      "path": "/downloads"
    },
    {
      "step": "writeFile",
      "path": "/downloads/plugin.zip",
      "data": {
        "resource": "url",
        "url": "https://raw.githubusercontent.com/WordPress/block-development-examples/deploy/zips/basic-esnext-a2ab62.zip",
        "caption": "Downloading plugin..."
      }
    },
    {
      "step": "installPlugin",
      "pluginZipFile": {
        "resource": "vfs",
        "path": "/downloads/plugin.zip"
      }
    },
    {
      "step": "runPHP",
      "code": "<?php require '/wordpress/wp-load.php'; wp_insert_post( array('post_title'    => 'Test block','post_content'  => '<div style=\"background-color:#900;color:#fff;padding:20px\" class=\"wp-block-block-development-examples-basic-esnext-a2ab62\">Hello World, step 1 (from the frontend).</div>', 'post_status'   => 'publish', 'post_author'   => 1 ) );"
    }
  ]
}
```

## Testing instructions

1. Preview the changes
2. Click the Live Demo link
3. Confirm it shows this view:

<img width="851" alt="CleanShot 2023-12-10 at 13 13 08@2x" src="https://github.com/WordPress/block-development-examples/assets/205419/2d030330-2394-4aea-949a-cd7bccb900c2">

Repeat for the "no build" example.

## Caveats

The `landingPage` link contains a hardcoded post ID right now. It's not ideal, as a future WordPress release could ship with different starter content and break that redirection. Playground will need to provide a mechanism for redirecting to a specific post, regardless of its relative ID.

I started an issue to track it here:

https://github.com/WordPress/blueprints/issues/52
